tests/v3/test_group.py

```diff
@@ -301,34 +301,28 @@ def test_group_contains(store: Store, zarr_format: ZarrFormat) -> None:
     assert "foo" in group
 
 
-def test_group_subgroups(store: Store, zarr_format: ZarrFormat) -> None:
-    """
-    Test the behavior of `Group` methods for accessing subgroups, namely `Group.group_keys` and `Group.groups`
-    """
+def test_group_child_iterators(store: Store, zarr_format: ZarrFormat):
```

Comment:
I decided to combine these tests since the setup is the same. And the test benefits from having both groups and arrays in the Group being accessed, since the code needs to filter to just child Groups / Arrays.